packages/graphql/src/mutations/uniswap/swapToToken.js

```diff
@@ -11,10 +12,20 @@ export async function swapToTokenTx(tokenValue) {
   const now = Math.round(+new Date() / 1000)
   const deadline = (block.timestamp < now - 60 ? now : block.timestamp) + 300
 
-  const value = await exchange.methods
+  const marketValue = await exchange.methods
     .getEthToTokenOutputPrice(tokenValue)
     .call()
 
+  // We send 1% extra value to cover 99% of market price moves.
+  // See https://github.com/OriginProtocol/origin/issues/2771
+  // for how this amount was determined.
+  //
+  // Marketplace prices change on average every 10 blocks.
+  // It takes some time to finish a transaction, and the exchange will
+  // fail our transaction if the user did not send enough eth for
+  // the new price.
+  const value = new BigNumber(marketValue).times(1.01).toFixed(0) // have to be an integer.
```

Comment:
I'm personally ok to use BigNumber since we are already using it in other places in our code on the back-end (bridge, growth, payout, ...).

@nick if you feel strongly we should use BN.js then the same result could be achieved by doing BN(marketValue).mul(101).div(100)